src/convertToObject.js

```diff
@@ -6,7 +6,26 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString.length < 1) {
+    return {};
+  }
+
+  const sourceElements = sourceString.split(';');
+
+  const sourceObject = sourceElements
+    .map((el) => el.trim())
+    .filter((el) => el.length > 0)
+    .reduce((obj, el) => {
+      const key = el.slice(0, el.indexOf(':')).trim();
+      const value = `${el.slice(el.indexOf(':') + 1, el.length).trim()}`;
+
+      return {
+        ...obj,
+        [key]: value,
+      };
```

Comment:
In your `reduce` callback, you are reconstructing a new object every time which is unnecessary and inefficient. Instead, simply assign the new key and value to the `obj` and return it. This will avoid the overhead of spreading the existing object into a new one on each iteration.